core/ROB.cpp

```diff
@@ -80,10 +69,20 @@ namespace olympia
     // An illustration of the use of the callback -- instead of
     // getting a reference, you can pull the data from the port
     // directly, albeit inefficient and superfluous here...
-    void ROB::robAppended_(const InstGroup &) {
-        for(auto & i : *in_reorder_buffer_write_.pullData()) {
-            reorder_buffer_.push(i);
-            ILOG("retire appended: " << i);
+    void ROB::robAppended_(const InstGroup &)
+    {
+        for (auto & i : *in_reorder_buffer_write_.pullData())
+        {
+            if (!i->isUOp())
+            {
+                // rob entries only for main instruction, uops aren't added.
+                reorder_buffer_.push(i);
+                ILOG("retire appended: " << i);
+            }
+            else
+            {
+                ILOG("UOp passed in, not adding to ROB " << i);
```

Comment:
For sanity, we could assert that the uop's parent instruction is the same as the youngest ROB entry.

Reply:
Added assertion in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)